evmrpc/block.go

```diff
@@ -154,7 +154,24 @@ func (a *BlockAPI) GetBlockReceipts(ctx context.Context, blockNrOrHash rpc.Block
 
 	// Get all tx hashes for the block
 	height := block.Block.Header.Height
-	txHashes := a.keeper.GetTxHashesOnHeight(a.ctxProvider(height), height)
+	txHashes := []common.Hash{}
+	for i, tx := range block.Block.Data.Txs {
+		sdkTx, err := a.txConfig.TxDecoder()(tx)
+		if err != nil {
+			fmt.Printf("error decoding tx %d in block %d, skipping\n", i, height)
+			continue
+		}
+		if len(sdkTx.GetMsgs()) == 0 {
+			continue
+		}
+		if evmTx, ok := sdkTx.GetMsgs()[0].(*types.MsgEVMTransaction); ok {
+			if evmTx.IsAssociateTx() {
+				continue
+			}
+			ethtx, _ := evmTx.AsTransaction()
+			txHashes = append(txHashes, ethtx.Hash())
+		}
+	}
```

Comment:
nit - shoudl we consider moving this to a helper function so it can be reused in filter.go?

Reply:
updated